client/components/domains/domain-registration-suggestion/index.jsx

```diff
@@ -28,32 +28,42 @@ const DomainRegistrationSuggestion = React.createClass( {
 
 	render() {
 		const { suggestion, translate } = this.props,
-			isAdded = hasDomainInCart( this.props.cart, suggestion.domain_name ),
+			domain = suggestion.domain_name,
+			isAdded = hasDomainInCart( this.props.cart, domain ),
 			domainFlags = [];
+
 		let buttonClasses, buttonContent;
 
-		if ( suggestion.domain_name ) {
+		if ( domain ) {
 			const newTLDs = [];
+			const a8cTLDs = [ '.ca', '.de', '.fr' ];
+			const tld = domain.substring( domain.indexOf( '.' ) );
 
-			if ( newTLDs.some(
-					( tld ) =>
-						endsWith( suggestion.domain_name, tld ) &&
-						suggestion.domain_name.substring( 0, suggestion.domain_name.length - ( tld.length + 1 ) ).indexOf( '.' ) === -1
-				) ) {
+			if ( includes( newTLDs, tld ) ) {
 				domainFlags.push(
 					<DomainSuggestionFlag
-						key={ `${ suggestion.domain_name }-new` }
+						key={ `${ domain }-new` }
 						content={ translate( 'New' ) }
 						status="success"
 					/>
 				);
 			}
+
+			if ( includes( a8cTLDs, tld ) ) {
+				domainFlags.push(
+					<DomainSuggestionFlag
+						key={ `${ domain }-a8c` }
+						content={ 'A8c' }
+						status="success"
```

Comment:
Good idea. Warning seems about right - we don't want people to think twice before selecting it, but they shouldn't feel like they should report it either.